docs/proposals/08-shoot-additional-container-runtimes.md

```diff
@@ -0,0 +1,139 @@
+
+Kubernetes supports running pods using these alternate runtimes via the [RuntimeClass](https://kubernetes.io/docs/concepts/containers/runtime-class/) concept, which was promoted to Beta in Kubernetes 1.14. Once Kubernetes is configured to use the Container Runtime Interface to control pods, it becomes possible to leverage CRI and run specific pods using different Runtime Classes. Additionally, configuring Kubernetes to use CRI instead of the legacy Dockershim is [faster](https://events19.linuxfoundation.org/wp-content/uploads/2017/11/How-Container-Runtime-Matters-in-Kubernetes_-OSS-Kunal-Kushwaha.pdf).
+
+The motivation behind this proposal is to make all of this functionality accessible to clusters managed by Gardener.
+
+### Goals
+
+* Gardener must default to configuring its managed clusters with the CRI interface instead of the legacy Dockershim. This change must be transparent to Gardener users and must simply work out-of-the-box for new and existing clusters.
+* Low-level runtimes like gVisor or Kata Containers are provided as gardener extensions which are (optionally) installed into a landscape by the Gardener operator. There must be no runtime-specific knowledge in the core Gardener code.
+* It shall be possible to configure multiple low-level runtimes in Gardener clusters, on the Worker Group level.
+
+### Non-Goals
+
+* Exposing the configuration of the underlying runtimes to the user is beyond the scope of this proposal. The initial version will support a default common-sense configuration for each of the underlying runtimes. If required, future enhancements will make it possible to fine-tune the underlying runtimes.
+
+## Proposal
+
+Gardener today assumes that all supported operating systems have Docker pre-installed in the base image. Starting with Docker Engine 1.11, Docker itself was [refactored](https://www.docker.com/blog/docker-engine-1-11-runc/) and cleaned-up to be based on the [containerd](https://containerd.io/) library. The first phase would be to change the default Kubelet configuration as described [here](https://kubernetes.io/docs/setup/production-environment/container-runtimes/#containerd) so that Kubernetes would use containerd instead of the default Dockershim. This will be implemented for CoreOS, Ubuntu, and SuSE-JeOS. 
+
+Once Gardener clusters are migrated to use CRI, we will implement two Gardener extensions, providing gVisor and Kata Containers as options for Gardener landscapes. The `WorkerGroup` specification will be extended to allow specifying a list of additional required Runtimes for nodes in that group. For example:
+```yaml
+workers:
+- name: worker-b8jg5
+  machineType: m5.large
+  volumeType: gp2
+  volumeSize: 50Gi
+  autoScalerMin: 1
+  autoScalerMax: 2
+  maxSurge: 1
+  containerRuntimes:
```

Comment:
I would change this to slice of objects so in the future this can be extended with additional configuration:

```yaml
  containerRuntimes:
  - name: gvisor
  - name: kata-containers
```